frontend/components/service/comboForm.js

```diff
@@ -1,347 +1,372 @@
+        setComboValue({...comboValue, [name]: value});
     };
-    request
-      .post(BuildPath('services'))
-      .send(data)
-      .set('Accept', 'application/json')
-      .then((res) => {
-        console.log(res.status);
-      })
-      .catch((err) => {
-        console.log(err);
-      });
-    closeClearValue();
-  };
-  const handleEditSubmit = () => {
-    //if all the madatory is not empty
-    let data = {
-      name: '',
-      serviceCode: '',
-      total_duration: '',
-      total_price: '',
-      service_ids: ['id', 'id'],
+    const removeService = (item) => {
+        handleServiceCheck(false, item);
+        if (serviceCheckList.length == 1) {
+            // the update of component is kind of late
+            closeClearValue();
+        }
     };
-    request
-      .post(BuildPath('services/combo/edit/id'))
-      .send(data)
-      .set('Accept', 'application/json')
-      .then((res) => {
-        console.log(res.status);
-      })
-      .catch((err) => {
-        console.log(err);
-      });
-    closeClearValue();
-  };
-  // add new service to the combo, pass the display only the service not yet in the combo
-  return (
-    <div>
-      <Dialog open={openDialog} fullWidth={true} maxWidth="lg" scroll="body">
-        <DialogTitle>
-          {isEdit() ? 'Edit Combo' : 'New Combo'}
-          <IconButton
-            aria-label="close"
-            onClick={closeClearValue}
-            size="medium"
-            sx={{
-              position: 'absolute',
-              right: 10,
-              top: 10,
-            }}
-          >
-            <Close />
-          </IconButton>
-        </DialogTitle>
-        <DialogContent>
-          <Stack spacing={2}>
-            <TextField
-              fullWidth
-              id="comboName"
-              value={comboName}
-              onChange={(event) => handleSetValue(event)}
-            />
-            {serviceCheckList &&
-              serviceCheckList.map((item) => (
-                <ComboItem
-                  key={item.id}
-                  item={item}
-                  handleServiceCheck={handleServiceCheck}
-                  removeService={removeService}
-                  isEdit={false}
-                />
-              ))}
-            {isEdit() ? (
-              <EditCombo
-                editDialog={editDialog}
-                setEditDialog={setEditDialog}
-                serviceCheckListAddable={serviceCheckList}
-                handleServiceCheck={handleServiceCheck}
-                removeService={removeService}
-                serviceCheckList={serviceCheckList}
-                handleSave={handleSave}
-              />
-            ) : null}
 
-            <div className={styled.flexAlignContainer}>
-              <label>Total Duration: </label>
-              <TextField
-                id="comboDuration"
-                value={(comboDuration / 3600000).toFixed(1)}
-                // type="number"
```

Comment:
remove this comment

Reply:
The same logic here, ( the pink color text are the lines that already removes compare to the previous one)